vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -100,7 +100,32 @@ public ImageData getImageDimensions(IVSImage image, int width, int height) {
     }  
     
     /**
-     * Method to return the total pages sufficient to display all images  
+     * Method to return the requested images
+     * 
+     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
+     *                page is returned
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List<IVSImage> getImages(int pageNo, String sortedBy, String order) {
+        Sort sortingParameters = Sort.by(SortByField.CREATION_DATE.getValue()).descending();
+        pageNo = validatePageNumber(pageNo);
+        if (sortedBy != null) {
+            if (order.contentEquals("desc")) {
+                sortingParameters = Sort.by(sortedBy).descending();
```

Comment:
you probably want to validagte `sortBy` to make sure it's a valid field. For now, maybe just have a hard-coded list of valid sortable fields.

Reply:
The method SortByField.getAllValues().contains(sortedBy) at line 117 does validate if the sortedBy value is one of the predefined values for sorting.

Reply:
yes, I believe my comment was for the previous version